migration/sql-files/103-update-non-root-iteration-path.sql

```diff
@@ -0,0 +1,3 @@
+-- update root iteration and area paths to use converted ids
+UPDATE iterations SET path=text2ltree(concat(path, concat('.',replace(cast(id as text), '-', '_')))) WHERE path!='' OR NOT NULL; 
```

Comment:
Is that `OR NOT NULL` really required here? In the previous query there is a constraint set which prevents null value for path column, right?

Reply:
@baijum it is required because the table allows for `NULL` values. See 

```
postgres=# \d iterations
                                 Table "public.iterations"
          Column          |           Type           |              Modifiers              
--------------------------+--------------------------+-------------------------------------
 created_at               | timestamp with time zone | 
 updated_at               | timestamp with time zone | 
 deleted_at               | timestamp with time zone | 
 id                       | uuid                     | not null default uuid_generate_v4()
 space_id                 | uuid                     | 
 path                     | ltree                    | 
 start_at                 | timestamp with time zone | 
 end_at                   | timestamp with time zone | 
 name                     | text                     | 
 description              | text                     | 
 state                    | iteration_state          | default 'new'::iteration_state
 relationships_changed_at | timestamp with time zone | 
 user_active              | boolean                  | not null default false
Indexes:
    "iterations_pkey" PRIMARY KEY, btree (id)
    "iterations_name_space_id_path_unique" UNIQUE CONSTRAINT, btree (space_id, name, path)
    "iteration_path_gist_idx" gist (path)
    "ix_name" btree (name)
    "ix_project_id" btree (space_id)
Foreign-key constraints:
    "iterations_space_id_spaces_id_fk" FOREIGN KEY (space_id) REFERENCES spaces(id) ON DELETE CASCADE
```

Reply:
In the `102-update-root-area-and-iteration-path-field.sql` script there is an alter statement to add a constraint which prevents null value for path column, right?

Reply:
@baijum @kwk when i combine both 102 and 103 i will change this